meta-lmp-base/classes/lmp.bbclass

```diff
@@ -3,7 +3,7 @@ FIO_PUSH_CMD ?= "fiopush"
 FIO_CHECK_CMD ?= "fiocheck"
 SOTA_TUF_ROOT_DIR ?= "usr/lib/sota/tuf"
 
-IMAGE_FSTYPES += "${@bb.utils.contains('EFI_PROVIDER', 'systemd-boot', 'ota-esp', ' ', d)}"
+IMAGE_FSTYPES:append:sota = "${@bb.utils.contains('EFI_PROVIDER', 'systemd-boot', ' ota-esp', '', d)}"
```

Comment:
There is only one problem here with append is that it will affect every image built by the build, including the initrds, so this won't really work without additional changes.

Reply:
Unless you also add ota-esp to the remove list, e.g.:

`meta-lmp-base/recipes-core/images/initramfs-ostree-lmp-image.bb:IMAGE_FSTYPES:remove = "wic wic.gz wic.bmap wic.nopt ext4 ext4.gz"`

But for intel there is also core-image.

Reply:
yeah this fixes the `lmp-base` but I have see this circular dependency side effect so it breaks `lmp` maybe something like:
```
IMAGE_FSTYPES += " \
    ${@bb.utils.contains('DISTRO_FEATURES', 'sota', \
        ${@bb.utils.contains('EFI_PROVIDER', 'systemd-boot', 'ota-esp', '', d)} \
    '', d)} \
"
```

Reply:
yeah, something along that line might work

Reply:
In the end I ended up adding a new variable IMAGE_FSTYPES_OTA_ESP for this propose. Teste with `lmp` and `lmp-base`.